gcc v7.2.0 warning [-Wformat-overflow=]

Compiling with gcc  v > 6.4 (I tested with 7.2.0) which seems more conservative by default, I got some warnings relative to  format-overflow. I not sure how serious this is. Overflow is never a good things in computer science :wink: 

>   warning: ‘sprintf’ may write a terminating nul past the end of the destination [-Wformat-overflow=]
warning: ‘ bytes’ directive writing 6 bytes into a region of size between 5 and 18 [-Wformat-overflow=]

Here with the context.

```
../../LASzip/src/laszip.cpp: In member function ‘bool LASzip::check_items(U16, const LASitem*, U16)’:
../../LASzip/src/laszip.cpp:274:6: warning: ‘ bytes’ directive writing 6 bytes into a region of size between 5 and 18 [-Wformat-overflow=]
 bool LASzip::check_items(const U16 num_items, const LASitem* items, const U16 point_size)
```

``` 
../../LASzip/src/laszip.cpp: In member function ‘bool LASzip::check(U16)’:
../../LASzip/src/laszip.cpp:294:6: warning: ‘ bytes’ directive writing 6 bytes into a region of size between 5 and 18 [-Wformat-overflow=]
 bool LASzip::check(const U16 point_size)
```

```
laswriter_las.cpp: In function ‘BOOL LASwriterLAS::open(ByteStreamOut*, const LASheader*, U32, I32, I32)’:
laswriter_las.cpp:126:6: warning: ‘__builtin___sprintf_chk’ may write a terminating nul past the end of the destination [-Wformat-overflow=]
```

Thanks

--- a/LASlib/src/laswriter_las.cpp
+++ b/LASlib/src/laswriter_las.cpp
@@ -718,7 +718,7 @@ BOOL LASwriterLAS::open(ByteStreamOut* stream, const LASheader* header, U32 comp
       fprintf(stderr,"ERROR: writing record_length_after_header %d\n", (I32)record_length_after_header);
       return FALSE;
     }
-    CHAR description[32] = "\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0";
+    CHAR description[33] = "\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0";
     sprintf(description, "tile %s buffer %s", (header->vlr_lastiling->buffer ? "with" : "without"), (header->vlr_lastiling->reversible ? ", reversible" : ""));
     if (!stream->putBytes((U8*)description, 32))
     {

--- a/LASzip/src/lasreadpoint.cpp
+++ b/LASzip/src/lasreadpoint.cpp
@@ -132,7 +132,7 @@ BOOL LASreadPoint::setup(U32 num_items, const LASitem* items, const LASzip* lasz
         readers_raw[i] = new LASreadItemRaw_GPSTIME11_LE();
       else
         readers_raw[i] = new LASreadItemRaw_GPSTIME11_BE();
-        break;
+       break;
     case LASitem::RGB12:
     case LASitem::RGB14:
       if (IS_LITTLE_ENDIAN())

--- a/LASzip/src/laszip.cpp
+++ b/LASzip/src/laszip.cpp
@@ -284,7 +284,7 @@ bool LASzip::check_items(const U16 num_items, const LASitem* items, const U16 po
   }
   if (point_size && (point_size != size))
   {
-    CHAR temp[64];
+    CHAR temp[66];
     sprintf(temp, "point has size of %d but items only add up to %d bytes", point_size, size);
     return return_error(temp);
   }

